broker/src/main/java/io/camunda/zeebe/broker/partitioning/PartitionStartup.java

```diff
@@ -0,0 +1,143 @@
+  private final RaftPartitionFactory raftPartitionFactory;
+  private final ZeebePartitionFactory zeebePartitionFactory;
+
+  PartitionStartup(
+      final ActorSchedulingService schedulingService,
+      final PartitionManagementService partitionManagementService,
+      final RaftPartitionFactory raftPartitionFactory,
+      final ZeebePartitionFactory zeebePartitionFactory) {
+    this.schedulingService = schedulingService;
+    this.partitionManagementService = partitionManagementService;
+    this.raftPartitionFactory = raftPartitionFactory;
+    this.zeebePartitionFactory = zeebePartitionFactory;
+  }
+
+  /**
+   * Bootstraps the partition with the given id. If the partition does not already exist locally, it
+   * will be initialized from static configuration. If either raft or Zeebe partition fails during
+   * startup, both will be stopped to avoid half-running partitions or leaked resources.
+   *
+   * @return future that completes successfully with the started partition when it is ready to use
+   *     or exceptionally if the partition fails to start.
+   */
+  CompletableFuture<StartedPartition> bootstrap(final PartitionMetadata partitionMetadata) {
+    final var bootstrap = new BootstrapActor(partitionMetadata);
+    schedulingService.submitActor(bootstrap);
+    return bootstrap.result();
+  }
+
+  record StartedPartition(RaftPartition raftPartition, ZeebePartition zeebePartition) {}
+
+  /**
+   * Orchestrates the bootstrap of a partition. This is a short-lived actor that starts the
+   * bootstrap on actor start and stops itself when the bootstrap is complete.
+   */
+  final class BootstrapActor extends Actor {
+
+    private static final Logger LOG =
+        LoggerFactory.getLogger("io.camunda.zeebe.broker.partitioning.bootstrap");
+    private final CompletableFuture<StartedPartition> result = new CompletableFuture<>();
+    private final PartitionMetadata partitionMetadata;
+    private final String name;
+
+    public BootstrapActor(final PartitionMetadata partitionMetadata) {
+      name = buildActorName("Bootstrap", partitionMetadata.id().id());
+      this.partitionMetadata = partitionMetadata;
+    }
+
+    @Override
+    protected Map<String, String> createContext() {
+      final var context = super.createContext();
+      context.put(ACTOR_PROP_PARTITION_ID, Integer.toString(partitionMetadata.id().id()));
+      return context;
+    }
+
+    @Override
+    public String getName() {
+      return name;
+    }
+
+    @Override
+    protected void onActorStarted() {
+      final var partitionId = partitionMetadata.id().id();
+      final var raftPartition = raftPartitionFactory.createRaftPartition(partitionMetadata);
+      // Call open on another thread because the raft partition will build the snapshot store which
+      // needs to wait for the snapshot store actor to be started. Synchronously waiting is not
+      // allowed from an actor, so delegate to another thread.
```

Comment:
:thought_balloon: This is also something worth refactoring. The snapshot store could be already built and provided to raft, instead of letting raft server creating it.